### PR TITLE
Review keyboardRect calculation

### PR DESCRIPTION
**Pull request by YGuerin**
GitHub Issue (If applicable): #

## PR Type
What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

On Android, Keyboard measurement can be negative when it's hidden

## What is the new behavior?

Keyboard measurement is limited to positive values only

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

https://nventive.visualstudio.com/Umbrella/_workitems/edit/145363
https://nventive.visualstudio.com/Umbrella/_workitems/edit/145370
https://nventive.visualstudio.com/Umbrella/_workitems/edit/145379

### CODE REVIEW

File: src/Uno.UI/UI/ViewManagement/InputPane/InputPane.cs [hunk 1]
```diff
@@ -84,7 +84,11 @@ public bool TryHide()
 
 		internal void OnOccludedRectChanged()
 		{
+#if __ANDROID__
```
Comment (MatFillion) [inline src/Uno.UI/UI/ViewManagement/InputPane/InputPane.cs]: it seems odd to me that we return KeyboardRect. For Android are we using OccludedRect at all then?

**Comment by jeromelaban** [comment]
This PR needs a UI Test

File: src/Uno.UI/UI/ViewManagement/InputPane/InputPane.cs (hunk 1 above)
Comment (agneszitte-nventive) [reply src/Uno.UI/UI/ViewManagement/InputPane/InputPane.cs]: Was thinking the same, NavigationBarRect is not needed at all for android for the inputpane ?

Comment (agneszitte-nventive) [reply src/Uno.UI/UI/ViewManagement/InputPane/InputPane.cs]: Was thinking the same, NavigationBarRect is not needed at all for android for the inputpane ?

Comment (YGuerin) [reply src/Uno.UI/UI/ViewManagement/InputPane/InputPane.cs]: I think you right. I will add some UI tests but it's probably the wrong place to fix it

Comment (agneszitte-nventive) [reply src/Uno.UI/UI/ViewManagement/InputPane/InputPane.cs]: Thanks in advance for the UI tests ;) !

Comment (XiaotianNetlift) [reply src/Uno.UI/UI/ViewManagement/InputPane/InputPane.cs]: Both the navigation bar and keyboard may be blocking the bottom area. Hence why they are both being propagated from here. OccludedRect should be used to calculate bottom offset. And, keyboard and navigation rects are used to check their visibility.

Comment (YGuerin) [reply src/Uno.UI/UI/ViewManagement/InputPane/InputPane.cs]: Indeed. As I said above, this is the wrong place to fix this bug.
I've found that NavigationRect is always equals to 48, regardless of navigation bar visibility (I've found that thanks to the UI Tests). I didn't have time to fix it this week but I hope I'll be able to do it the next one.

**Comment by YGuerin** [comment]
Creating a new pull request for that one.
https://github.com/nventive/Uno/pull/674

**Pull request closed without merging**